lib/src/navigation.dart

```diff
@@ -127,6 +129,20 @@ class BottomNavScaffold extends ConsumerWidget {
   Widget build(BuildContext context, WidgetRef ref) {
     final currentTab = ref.watch(currentBottomTabProvider);
 
+    ref.listen(appLinksProvider, (_, appLinkUri) {
+      switch (appLinkUri) {
+        case AsyncData(:final Uri value):
```

Comment:
You can do this if you want to match only non null value
```suggestion
        case AsyncData(:final value?):
```

Reply:
huh, are you sure? I though `case AsyncData(:final Uri value)` would match `if and only if` the `uri` is not `null`, while `case AsyncData(:final value?)` would also match the `null` value, so you'd need an additional `if (uri != null)` check in that case

Reply:
Yes the notation also leaves me perplex but it is documented [here](https://dart.dev/language/pattern-types#null-check).

Both solution works but you don't have to specify the type with the null check pattern type.

Reply:
Ok wow, that is really unintuitive, especially when coming from other languages